themes/bootstrap3/templates/search/results-list.phtml

```diff
@@ -0,0 +1,18 @@
+<?php
+  // Create a css class for results from the search class:
+  $searchClass = $searchClassCss = $this->params->getSearchClassId();
+  if (!ctype_upper($searchClassCss)) {
+    $searchClassCss = preg_replace('/([a-zA-Z])(?=[A-Z])/', '$1-', $searchClassCss);
+  }
+  $this->resultsClass = 'search-results-' . strtolower($searchClassCss);
+  // Add any additional classes and attributes:
+  if ($this->results->getOptions()->loadResultsWithJsEnabled()) {
+    $this->resultsClass .= ' js-record-list';
+    if ($this->resultsAttrs) {
+      $this->resultsAttrs->add('data-backend', $searchClass);
```

Comment:
We could check the type, but I'm not sure what would be a useful course of action if it's a string.

Reply:
At least three options:

1.) Is there a way to initialize the htmlAttributes helper from a string of attributes? If so, we could promote strings to objects.

2.) We could concatenate the attribute to the string if it's a string.

3.) We could throw an exception, which is probably better than a low-level PHP error.

Since we treat resultsAttrs like a string in the list-*.phtml templates, it's not immediately obvious that we're using the htmlAttributes helper, depending on how you trace the code for the property, so I think it's probably worth doing _something_, though not necessarily anything that requires tons of time/effort.

Reply:
Alas, it can't be built from a string. I added a type check now since that keeps the actual code clean.

Reply:
Maybe a mechanism for populating htmlAttributes from a string would be a useful future enhancement to that helper, but for now this does the job!

Reply:
Yes, but since the idea of the helper is to avoid hand-crafting attribute strings, it kind of makes sense to not support population from a string. :)

Reply:
Fair point. :-)